airbyte-integrations/connectors/source-google-sheets/google_sheets_source/helpers.py

```diff
@@ -199,7 +199,7 @@ def row_contains_relevant_data(cell_values: List[str], relevant_indices: Iterabl
 
     @staticmethod
     def get_spreadsheet_id(id_or_url: str) -> str:
-        if re.match(r"(http://)|(https://)", id_or_url):
+        if re.match(r"(https://)", id_or_url):
```

Comment:
@alafanechere I think this is an example where us scanning the source code falls down.

For example, like here, a source "MAY" need to check the contents of datapoint in a source for HTTP. We shouldn't prevent them from doing that.

Reply:
I don't really get why a source may need to check the contents of an HTTP data point and why we shouldn't prevent them from doing that.

Reply:
Fair question! and a fair point. Let me give an example.

_**Warning Im going down a "might happen" path. Feel free to call out as this wont happen.**_

So say for example I have a connector for a product that can be self hosted. Think matterport, grafana, strapi, your own service.

In the `spec` for the connector I ask for a required field of `apiUrl`.

Because I'm self hosted I know that a customer could have airbyte-oss and my-self-hosted-product on the same machine. Meaning that the value of `apiUrl` could be either `http://localhost:3333/api` or `https://org.my-self-hosted-product.com/api`

I may look for that http definition to either Fail and say http is not supported for this connector unless its localhost

But I dont want to take this too far with "What ifs". So Ill throw an approve on this since you know the product better than I do, and merge if you think this is a non-issue 🙂

Reply:
I agree that this scenario can happen. What you mention exists for the `source-file` connector, which I ignored from the QA check. For google sheets this is unlikely to happen. I'd prefer us to be aware of which connector behaves like you mention and ignore them in the check if needed.